core/sql/executor/ex_sort.cpp

```diff
@@ -330,21 +364,39 @@ ExSortTcb::~ExSortTcb()
    nfDiags_ = NULL;
   
   freeResources();
-  
-  if(pool_)
-    delete pool_;
+
 };
   
 ////////////////////////////////////////////////////////////////////////
 // Free Resources
 //
 void ExSortTcb::freeResources()
 {
-  if (sortPool_)
+  if (partialSortPool_)
+  {
+    delete partialSortPool_;
+    partialSortPool_ = NULL;
+  }
+  if (regularSortPool_)
+  {
+    delete regularSortPool_;
+    regularSortPool_ = NULL;
+  }
+  if (topNSortPool_)
+  {
+    delete topNSortPool_;
+    topNSortPool_ = NULL;
+  }
+  if (sortPool_ && (sortPool_ != receivePool_))
```

Comment:
What if sortPool_ is not null and  sortPool_ == receivePool_? Shouldn't we set it to NULL in this case (without doing a delete)?